### ISSUE TITLE
azurerm_logic_app_standard - virtual_network_subnet_id not exposed as it should

### ISSUE THREAD

**Issue by lbergeron01** [hashicorp/terraform-provider-azurerm#18376]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.22.0

### Affected Resource(s)/Data Source(s)

azurerm_logic_app_standard

### Terraform Configuration Files

```hcl
resource "azurerm_logic_app_standard" "logic_app_standard" {
  name                = var.name
  location            = var.location
  resource_group_name = var.resource_group_name

  app_service_plan_id        = var.service_plan_id
  storage_account_name       = var.storage_account_name
  storage_account_access_key = var.storage_account_access_key
  virtual_network_subnet_id  = data.azurerm_subnet.vnet_integration_subnet.id
}

data "azurerm_subnet" "vnet_integration_subnet" {
  name                 = var.app_service_vnet_integration.private_subnet_name
  virtual_network_name = var.app_service_vnet_integration.virtual_network_name
  resource_group_name  = var.app_service_vnet_integration.virtual_network_resource_group_name
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: Unsupported argument
│
│   on .terraform\modules\infra.aircanadastock_logic_app\logic-app\logic-app-standard\logic-app-standard-main.tf line 100, in resource "azurerm_logic_app_standard" "logic_app_standard":
│  100:   virtual_network_subnet_id  = data.azurerm_subnet.vnet_integration_subnet[count.index].id
│
│ An argument named "virtual_network_subnet_id" is not expected here.
╵
```

### Expected Behaviour

As documented in [https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_virtual_network_swift_connection](app_service_virtual_network_swift_connection), azurerm_logic_app_standard should support virtual_network_subnet_id.

Looks like PR #17734 updated documentation, but issue #17731 has not been merged into the code base 3.16.0 like other issues with virtual_subnet_network_id (#17576, #17572, #17494, #17584).

### Actual Behaviour

```
╷
│ Error: Unsupported argument
│
│   on .terraform\modules\infra.aircanadastock_logic_app\logic-app\logic-app-standard\logic-app-standard-main.tf line 100, in resource "azurerm_logic_app_standard" "logic_app_standard":
│  100:   virtual_network_subnet_id  = data.azurerm_subnet.vnet_integration_subnet[count.index].id
│
│ An argument named "virtual_network_subnet_id" is not expected here.
╵
```

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_virtual_network_swift_connection

#17734
#17731

**Comment by sinbai** [comment]
@lbergeron01 thanks for raising this issue here. I have reminded the reviewer of PR #17731. Hope this helps speed up the PR merging.

**Comment by asyednl** [comment]
I'm still getting this message? do we have an eta on this?

**Comment by sinbai** [comment]
It seems the functionality has been released in [v3.23.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.23.0/CHANGELOG.md). Could you try this version?

**Comment by rcskosir** [comment]
Thanks for opening this issue. A related PR (#17731) has been merged and was released in v3.23.0. Please upgrade to that version or the latest, thanks!
